emuu-client/src/components/CreatorsPage/CreatorsPage.js

```diff
@@ -1,83 +1,308 @@
+  );
+  const [autocompleteState, setAutocompleteState] = useState({});
+  const [searchInput, setSearchInput] = useState("");
+  const [count, setCount] = useState(0);
+  const [subscriberActionCount, setSubsciberActionCount] = useState(0);
+  const [updatedSubscribersList, setUpdateSubscribersList] = useState([]);
 
-  useEffect(async () => {
-    if (video) {
-      localStorage.setItem("video", JSON.stringify(video));
+  const firebaseData = JSON.parse(localStorage.getItem("firebase-data"));
+
+  const autocomplete = useMemo(
+    () =>
+      createAutocomplete({
+        onStateChange({ state }) {
+          setAutocompleteState(state);
+          setSearchInput(state.query);
+          if (count === 0) {
+            setCount((count) => count + 1);
+          }
+        },
+        getSources() {
+          return [
+            {
+              sourceId: "pages-source",
+              getItemInputValue({ item }) {
+                // search item
+                return item.query;
+              },
+              getItems({ query }) {
+                // takes your search input and checks if anything that matches it exists in your dataset
+                if (!query) {
+                  return firebaseData;
+                }
+                return firebaseData.filter(
+                  (item) =>
+                    item.VideoTitle?.toLowerCase().includes(
+                      query.toLowerCase()
+                    ) ||
+                    item.Username?.toLowerCase().includes(
+                      query.toLocaleLowerCase()
+                    )
+                );
+              },
+              templates: {
+                item({ item }) {
+                  return item.VideoTitle || item.Username;
+                },
+              },
+            },
+          ];
+        },
+      }),
+    [count]
+  );
+
+  const dataSet = autocompleteState?.collections?.[0]?.items;
+  const searchResultsVideosArr = dataSet?.filter(
+    (obj) => obj.hasOwnProperty("VideoUrl") && obj.hasOwnProperty("Username")
+  );
+  const searchResultsUsersArr = dataSet?.filter(
+    (obj) => !obj.hasOwnProperty("VideoUrl") && obj.hasOwnProperty("Username")
+  );
+  const showSearchResults =
+    searchResultsVideosArr?.length > 0 || searchResultsUsersArr?.length > 0;
+
+  const userName = localStorage.getItem("displayName");
+
+  const usersArr = firebaseData.filter(
+    (obj) => obj.hasOwnProperty("Username") && !obj.hasOwnProperty("VideoUrl")
+  );
+  const videosArr = firebaseData.filter(
+    (obj) => obj.hasOwnProperty("Username") && obj.hasOwnProperty("VideoUrl")
+  );
+
+  const handleCreatorProfile = (creatorsName) => {
+    const creatorsData = usersArr.filter(
+      (user) => user.Username === creatorsName
+    );
+    const creatorsDataVideos = videosArr.filter(
+      (video) => video.Username === creatorsName
+    );
+    localStorage.setItem("creatorsData", JSON.stringify(creatorsData));
+    localStorage.setItem(
```

Comment:
Why store the videos locally on the creators page? We're pulling from Firebase (migrated to backend with video API) and displaying it.